src/components/FilterSearch.tsx

```diff
@@ -102,8 +102,9 @@ export function FilterSearch({
       }
       answersActions.setFilterOption({ ...newFilter, displayName: newDisplayName, selected: true });
       setCurrentFilter(newFilter);
-      answersActions.setOffset(0);
       if (select && searchOnSelect) {
+        answersActions.setOffset(0);
+        answersActions.resetFacets();
```

Comment:
if searchOnSelect is false, would FilterSearch rely on ApplyFiltersButton to execute a search? That would only call setOffset but not resetFacets

Reply:
Yes, in that case it would rely on the search being triggered from the search bar, `ApplyFiltersButton`, etc. In the case of the `ApplyFiltersButton`, it wouldn't make sense to reset the facets, because the user may be trying to apply new facets. Product only wants the resetting facets behavior when `searchOnSelect` is true because that's something the user can't change easily. They'll be adding docs for how to get the same behavior when `searchOnSelect` is false (e.g. creating a search button for filter search and having it call `resetFacets`)